src/bank.rs

```diff
@@ -828,4 +869,38 @@ mod tests {
         bank.process_ledger(genesis.chain(block)).unwrap();
         assert_eq!(bank.get_balance(&mint.pubkey()), 1);
     }
+
+    #[test]
+    fn test_rotate_vector() {
+        let test = vec![4, 1, 2, 3];
+        let etest = vec![4, 1, 2, 3];
+
+        let (left, right) = etest.split_at(1);
+        let mut right = right.to_vec();
```

Comment:
One less malloc: `right.chain(left).to_vec()`

Reply:
will have a go

Reply:
ug, fighting with types in rust to get chain() to work on right...

Reply:
```
878 |         let right = right.chain(left).to_vec();
    |                           ^^^^^
    |
    = note: the method `chain` exists but the following trait bounds were not satisfied:
            `&mut &[{integer}] : std::iter::Iterator`
            `&mut [{integer}] : std::iter::Iterator`
```

Reply:
Untested, but the chain docs suggest you need an `iter()`:
`right.iter().chain(left).to_vec()`